commit 987905c8915c102ffe8012d4a24c1ec1a73225cf
Author: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date:   2018-02-26T17:13:11+01:00 (1 year, 7 months ago)
    
    OS-6416 add bhyve UEFI firmware to illumos-extra [path to iasl]

diff --git a/Makefile b/Makefile
index 165f999..7692a74 100644
--- a/Makefile
+++ b/Makefile
@@ -73,6 +73,7 @@ SUBDIRS = \
 	screen \
 	socat \
 	tun \
+	uefi-edk2 \
 	uuid \
 	vim \
 	wget \
diff --git a/manifest b/manifest
index c40d74b..e2568f6 100644
--- a/manifest
+++ b/manifest
@@ -3022,7 +3022,7 @@ f usr/share/man/man1/screen.1 0444 root bin
 f usr/bin/socat 0555 root bin
 f usr/share/man/man1/socat.1 0444 root bin
 # uefi-edk2
-# f usr/share/bhyve/uefi-csm-rom.bin 0644 root bin
+f usr/share/bhyve/uefi-csm-rom.bin 0644 root bin
 # uuid
 f usr/bin/uuid 0755 root bin
 f usr/share/man/man1/uuid.1 0644 root bin
diff --git a/uefi-edk2/Makefile b/uefi-edk2/Makefile
index 49af187..e582e64 100644
--- a/uefi-edk2/Makefile
+++ b/uefi-edk2/Makefile
@@ -43,6 +43,7 @@ all: $(VER.64)/$(UNPACK_SENTINEL)
 	ln -sf $(STRAPPROTO)/usr/gnu/bin/gar $(VER.64)/Build/ar
 	ln -sf $(STRAPPROTO)/usr/gnu/bin/gobjcopy $(VER.64)/Build/objcopy
 	ln -sf /opt/local/bin/nasm $(VER.64)/Build/nasm
+	ln -sf $(STRAPPROTO)/iasl $(VER.64)/Build/iasl
 	$(MAKE) $(MAKE_ARGS) ARCH=$(ARCH) -C $(VER.64)/BaseTools
 	bash -c "cd $(VER.64); source edksetup.sh; $(MAKE) $(MAKE_ARGS) ARCH=$(ARCH) -C BhyvePkg/Csm/BhyveCsm16"
 	bash -c "cd $(VER.64); source edksetup.sh; export ILGCC_BIN=$(PWD)/$(VER.64)/Build/; build -t ILGCC -a $(ARCH) -b $(UEFI_TARGET) -p BhyvePkg/BhyvePkgX64.dsc -DDEBUG_ON_SERIAL_PORT=TRUE -DFD_SIZE_2MB -DCSM_ENABLE=TRUE"
diff --git a/uefi-edk2/Patches/0001-Add-ILGCC-target-to-build-on-illumos.patch b/uefi-edk2/Patches/0001-Add-ILGCC-target-to-build-on-illumos.patch
index db771e9..95420fc 100644
--- a/uefi-edk2/Patches/0001-Add-ILGCC-target-to-build-on-illumos.patch
+++ b/uefi-edk2/Patches/0001-Add-ILGCC-target-to-build-on-illumos.patch
@@ -53,7 +53,7 @@ index 3fb1216..1b9099e 100644
 +
 +*_ILGCC_*_MAKE_PATH			= DEF(ILGCC_PREFIX)make
 +*_ILGCC_*_*_DLL				= ENV(GCC44_DLL)
-+*_ILGCC_*_ASL_PATH			= DEF(UNIX_IASL_BIN)
++*_ILGCC_*_ASL_PATH			= DEF(ILGCC_PREFIX)iasl
 +
 +*_ILGCC_*_OBJCOPY_PATH			= DEF(ILGCC_PREFIX)objcopy
 +*_ILGCC_*_CC_PATH			= DEF(ILGCC_PREFIX)gcc
